modules/es-cluster-health-is-red.adoc

```diff
@@ -0,0 +1,153 @@
+// Module included in the following assemblies:
+//
+// * logging/logging_alerts/troubleshooting-logging-alerts.adoc
+
+:_content-type: PROCEDURE
+[id="es-cluster-health-is-red_{context}"]
+= Elasticsearch cluster health status is red
+
+At least one primary shard and its replicas are not allocated to a node. Use the following procedure to troubleshoot this alert.
+
+.Procedure
+
+. Check the Elasticsearch cluster health and verify that the cluster `status` is red by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- health
+----
+
+. List the nodes that have joined the cluster by running the following command:
++
+[source,terminal]
+----
+$ oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_cat/nodes?v
+----
+
+. List the Elasticsearch pods and compare them with the nodes in the command output from the previous step, by running the following command:
++
+[source,terminal]
+----
+$ oc -n openshift-logging get pods -l component=elasticsearch
+----
+
+. If some of the Elasticsearch nodes have not joined the cluster, perform the following steps.
+
+.. Confirm that Elasticsearch has an elected control plane node by running the following command and observing the output:
```

Comment:
```suggestion
.. Confirm that Elasticsearch has an elected master node by running the following command and observing the output:
```

Reply:
Per https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#using-conscious-language we're supposed to use "control plane node" now.

Reply:
This is not specific to Kubernetes, it's something that's implemented in many products at Red Hat and is a requirement AFAIK for docs in the OpenShift repo. Since this was preexisting content, I am reluctant to change it to "master", especially because I think it would be pushed back on from the core OCP team.

Reply:
@kalexand-rh any thoughts on this, is `master` permitted in cases like this?

Reply:
I am not very familiar with Elasticsearch, but from the way it's written, it certainly sounds like it's a node joining an OpenShift cluster, in which case "master" should not be used. If it's an Elasticsearch node joining an Elasticsearch cluster, I suggest specifying "an Elasticsearch cluster" and trying to rewrite the sentence to avoid the term "master." If it's truly the only way to describe it, I think it's reluctantly allowed - we're cleaning up terminology, but I think that clarity is more important. (In other collections, I see "master" used with respect to Apache master nodes.) @bergerhoffer, is there a different way to approach this that we should consider?

Reply:
@kalexand-rh @abrennan89 I am not very familiar with Elasticsearch either.

In general, we should be avoiding the use of "master" if at all possible. The only real cases I'm aware of where we need to keep using it is if it's part of the code/config option (e.g. the `master` role), or is a part of some other non-RH technology/component that uses "master" as part of its terminology.

I can see in looking in the Elasticsearch documentation that they do seem to use the term "master node" throughout. I don't know enough about Elasticsearch to know if they're talking about their own node somewhere outside of the OCP cluster (in which case I'd say that we'd need to keep that term "master node") or if we really are talking about an OCP node in our OCP cluster (in which case I would go with "control plane node").

Another alternative would be to try to rewrite to avoid "master node" looking like we're talking about an OCP control plane node (if we aren't). Maybe by putting Elasticsearch in front of "node" or talking about the node with the `master` role. Not sure what would keep the accuracy here though.

Sorry to not be more helpful. But at the end of the day, we try to avoid it if we can write it another way, but if that really is the term that needs to be used, then we can use it.

Reply:
A collection of Elasticsearch instances (nodes) is called an Elasticsearch cluster, this is regardless where it is installed, it could be locally on a single machine or on the cloud.

The term "master node" is used throughout ES to indicate an _Elasticsearch_ (not OCP) node that has permissions to create and delete indices. [[ref]](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/modules-node.html#master-node). Maybe it is worth it to clarify this at the beginning of the section.

Reply:
The purpose of this document is to help customers debug issues with Elasticsearch. While I understand why the effort to use more inclusive language exists, I think in this case the usefulness of using the same vocabulary as the product you're debugging outweighs the language concern.

For example, this is the output of the `_cat/nodes` query above:

```
$ oc exec -n openshift-logging -c elasticsearch elasticsearch-cdm-vvnyug3j-1-577b77ccbf-l7vbg -- es_util --query=_cat/nodes?v
ip           heap.percent ram.percent cpu load_1m load_5m load_15m node.role master name
10.128.0.182           50          96   2    5.85    3.51     2.66 mdi       *      elasticsearch-cdm-vvnyug3j-3
10.128.0.181           48          96   2    5.85    3.51     2.66 mdi       -      elasticsearch-cdm-vvnyug3j-2
10.128.0.180           61          93   2    5.85    3.51     2.66 mdi       -      elasticsearch-cdm-vvnyug3j-1
```

The column the user should look at is called `master` and this also carries forward to the endpoint (`_cat/master`) used below. The words "control plane node" do not have any meaning in the context of Elasticsearch.

Reply:
That makes sense @xperimental , I will update it to master node